nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/UpdaterWorker.cs

```diff
@@ -101,7 +104,40 @@ public async Task RunAsync(string repoRootPath, string workspacePath, string dep
         string newDependencyVersion,
         bool isTransitive)
     {
-        _logger.Log($"Running for project [{projectPath}]");
+        _logger.Log($"Running for project file [{Path.GetRelativePath(repoRootPath, projectPath)}]");
+        if (!File.Exists(projectPath))
+        {
+            _logger.Log($"File [{projectPath}] does not exist.");
+            return;
+        }
+
+        var projectFilePaths = MSBuildHelper.GetProjectPathsFromProject(projectPath);
+        foreach (var projectFullPath in projectFilePaths.Concat([projectPath]))
+        {
+            // If there is some MSBuild logic that needs to run to fully resolve the path skip the project
+            if (File.Exists(projectFullPath))
+            {
+                await RunUpdaterAsync(repoRootPath, projectFullPath, dependencyName, previousDependencyVersion, newDependencyVersion, isTransitive);
```

Comment:
This is the heart of the change. We now collect referenced projects and update them as we do for dirs.proj.